### PR TITLE
Scaffolder: Add component name warning

### PR DESCRIPTION
**Pull request by nilsstre**
Added pattern to docs-template. 

Closes: #4216

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
image

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4400-7ej3ivecpaaunz6xsnzomrdj7vl6j4aq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d1dc5353bbb76eba0817b

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4400-7ej3ivecpaaunz6xsnzomrdj7vl6j4aq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d1dc5353bbb76eba0817b

File: .changeset/fifty-experts-exist.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': patch
```
Comment (Rugvip) [inline .changeset/fifty-experts-exist.md:2]: No need for a changeset here actually, the sample templates aren't part of the release 😅

**Review by Rugvip — APPROVED** [review]
Thanks! Good to merge after removing the changeset 👍

File: .changeset/fifty-experts-exist.md (hunk 1 above)
Comment (nilsstre) [reply .changeset/fifty-experts-exist.md:2]: Good to know, I will remove it.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4400-7ej3ivecpaaunz6xsnzomrdj7vl6j4aq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d1dc5353bbb76eba0817b

**Pull request merged**